eng/crossgen-comparison-job.yml

```diff
@@ -0,0 +1,176 @@
+        downloadType: single
+        artifactName: ${{ format('{0}_{1}_{2}_build', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
+        downloadPath: $(System.ArtifactsDirectory)
+
+
+    # Populate Product directory
+    - task: CopyFiles@2
+      displayName: Populate Product directory
+      inputs:
+        sourceFolder: $(System.ArtifactsDirectory)/${{ format('{0}_{1}_{2}_build', parameters.osIdentifier, parameters.archType, parameters.buildConfig) }}
+        contents: '**'
+        targetFolder: $(productDirectory)/$(targetFlavor)
+
+
+    # Create directories and ensure crossgen is executable
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - script: |
+          chmod +x $(productDirectory)/$(targetFlavor)/$(hostArchType)/crossgen
+          mkdir -p $(binDirectory)/Logs/$(crossFlavor)
+        displayName: Create directories and ensure crossgen is executable
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - script: |
+          mkdir $(binDirectory)\Logs\$(crossFlavor)
+        displayName: Create directories and ensure crossgen is executable
+
+
+    # Populate Core_Root
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - script: ./build-test.sh $(buildConfig) $(archType) $(crossArg) generatelayoutonly
+        displayName: Populate Core_Root
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - script: build-test.cmd $(buildConfig) $(archType) generateLayoutOnly
+        displayName: Populate Core_Root
+
+
+    # Create baseline output on the host (x64) machine
+    - task: PythonScript@0
+      displayName: Create cross-platform crossgen baseline
+      inputs:
+        scriptSource: 'filePath'
+        scriptPath: $(Build.SourcesDirectory)/tests/scripts/crossgen_comparison.py
+        ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+          arguments:
+            crossgen_framework
+            --crossgen   $(productDirectory)/$(targetFlavor)/$(hostArchType)/crossgen
+            --il_corelib $(productDirectory)/$(targetFlavor)/IL/System.Private.CoreLib.dll
+            --core_root  $(binDirectory)/tests/$(targetFlavor)/Tests/Core_Root
+            --result_dir $(binDirectory)/Logs/$(crossFlavor)
+        ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+          arguments:
+            crossgen_framework
+            --crossgen   $(productDirectory)\$(targetFlavor)\$(hostArchType)\crossgen
+            --il_corelib $(productDirectory)\$(targetFlavor)\IL\System.Private.CoreLib.dll
+            --core_root  $(binDirectory)\tests\$(targetFlavor)\Tests\Core_Root
+            --result_dir $(binDirectory)\Logs\$(crossFlavor)
+
+
+    # Send payload to Helix where the native output is generated and compared to the baseline
+    - template: /eng/common/templates/steps/send-to-helix.yml
+      parameters:
+        DisplayNamePrefix: Run native crossgen and compare output to baseline
+        HelixSource: 'pr/dotnet/coreclr/$(Build.SourceBranch)'
```

Comment:
I think `HelixSource` should start with `ci/dotnet/coreclr/` if it was triggered in CI